lib/logic/list-controller.js

```diff
@@ -53,60 +46,60 @@ ListController.prototype.getOpenPullRequests = function() {
-        unassigned: [],
-        awaitingReview: [],
-        awaitingAction: [],
-        reviewed: [],
-        lgtmed: []
-    };
-    for (var i = 0; i < prs.length; ++i) {
-        var pr = prs[i];
-        if (!pr.getAssignee()) {
-            pulls.unassigned.push(pr);
-        }
-        else if (pr.getComments().length <= 1 && pr.getActions().length === 0) {
-            // Allow 1 comment because creating the PR leaves a comment
-            pulls.awaitingReview.push(pr);
+        owned: {
+            unassigned: [],
+            awaitingReview: [],
+            awaitingAction: [],
+            reviewed: [],
+            lgtmed: []
+        },
+        assigned: {
+            needsAction: [],
+            othersInvolved: [],
+            noAction: []
+        },
+        mentioned: {
+            all: []
         }
-        else if (pr.getActionsToDo().length > 0) {
-            pulls.awaitingAction.push(pr);
-        }
-        else if (pr.isLGTM()) {
-            pulls.lgtmed.push(pr);
-        }
-        else {
-            pulls.reviewed.push(pr);
-        }
-    }
-    return pulls;
-};
-
-ListController.prototype._loadAssignedPulls = function(prs) {
-    var assignedPulls = {
-        assignedNeedsAction: [],
-        assignedOthersInvolved: [],
-        assignedNoAction: []
     };
     for (var i = 0; i < prs.length; ++i) {
         var pr = prs[i];
-        var me = pr.getAssignee();
-        if (pr.getOwner().equals(me)) {
-            // PR will appear in my pulls, don't duplicate it in assigned pulls
-            continue;
-        }
-        else if (pr.getOwner().equals(pr.getLastCommenter()) && pr.getActionsToDo().length === 0) {
-            assignedPulls.assignedNeedsAction.push(pr);
+        if (me.equals(pr.getOwner())) {
+            if (!pr.getAssignee()) {
+                pulls.owned.unassigned.push(pr);
+            }
+            else if (pr.getComments().length <= 1 && pr.getActions().length === 0) {
+                // Allow 1 comment because creating the PR leaves a comment
+                pulls.owned.awaitingReview.push(pr);
+            }
+            else if (pr.getActionsToDo().length > 0) {
+                pulls.owned.awaitingAction.push(pr);
+            }
+            else if (pr.isLGTM()) {
+                pulls.owned.lgtmed.push(pr);
+            }
+            else {
+                pulls.owned.reviewed.push(pr);
+            }
         }
-        else if (!me.equals(pr.getLastCommenter())) {
-            assignedPulls.assignedOthersInvolved.push(pr);
+        else if (me.equals(pr.getAssignee())) {
+            if (pr.getOwner().equals(pr.getLastCommenter()) && pr.getActionsToDo().length === 0) {
+                pulls.assigned.needsAction.push(pr);
+            }
+            else if (!me.equals(pr.getLastCommenter())) {
+                pulls.assigned.othersInvolved.push(pr);
+            }
+            else {
+                pulls.assigned.noAction.push(pr);
+            }
```

Comment:
All this logic I feel may be better if it was contained inside the `PullRequest` class and instead modified a `myStatus` prop. I think this for two main reasons:
 - A PR can only be in one of the arrays in `pulls` but this is not clear. If the pull request could only have 1 of N statuses that makes it clear that it can't exist in multiple forms.
 - To work out which array the PR needs to be in only involves inspecting itself and `me`.

This would also make it a lot easier for you to then pass around a list of PRs without having to drag a huge record of PRs which contains vital contextual information from the output of this function.